src/CookiePhysicsComponent.cpp

```diff
@@ -39,15 +45,46 @@ void CookiePhysicsComponent::updatePhysics(float deltaTime) {
 
     //TODO(nurgan) make the cookie "spin" when it is in the air
 
-    // If we hit anything, stop "forward" movement
-    GameObjectType objTypeHit = world.checkCollision(holder_);
+    // If we hit anything, stop "forward"
+    GameObject* objHit = world.checkCollision(holder_);
+    GameObjectType objTypeHit = objHit->type;
 
     if (objTypeHit == GameObjectType::STATIC_OBJECT || objTypeHit == GameObjectType::DYNAMIC_OBJECT) {
-            holder_->velocity = 0.0f;
+
+        BoundingBox objBB = objHit->boundBox;
+        BoundingBox cookieBB = holder_->boundBox;
+
+        glm::vec3 n;
+
+        //check on which side of the bounding box of the object the cookie hit and create normal for reflection
+        if((objBB.min_.x - epsilon <= cookieBB.max_.x && objBB.min_.x + epsilon >= cookieBB.max_.x) ||
+                (objBB.max_.x - epsilon <= cookieBB.min_.x && objBB.max_.x + epsilon >= cookieBB.min_.x)) {
+            n = glm::vec3(1.0, 0.0, 0.0);
+        }
+        if((objBB.min_.y - epsilon <= cookieBB.max_.y && objBB.min_.y + epsilon >= cookieBB.max_.y) ||
+                (objBB.max_.y - epsilon <= cookieBB.min_.y && objBB.max_.y + epsilon >= cookieBB.min_.y)){
+            n = glm::vec3(0.0, 1.0, 0.0);
+        }
+        if((objBB.min_.z - epsilon <= cookieBB.max_.z && objBB.min_.z + epsilon >= cookieBB.max_.z) ||
+                (objBB.max_.z - epsilon <= cookieBB.min_.z && objBB.max_.z + epsilon >= cookieBB.min_.z)){
+            n = glm::vec3(0.0, 0.0, 1.0);
+        }
+
```

Comment:
nice, we'll probably want this as a more generalized function for stuff like player hit collision, but for now this is good stuff